### ISSUE TITLE
Console error "closure invoked recursively or destroyed already" when setting signal checked in the conditional.

### ISSUE THREAD

**Issue by godstail** [sycamore-rs/sycamore#577]
In example component every click on the button produces console error:
```rust
#[component]
fn MyComponent<G: Html>(cx: Scope) -> View<G> {
    let bool_sig = create_signal(cx, false);
    view! { cx,
        button(on:click = |_| {
            bool_sig.set(true);
        }) {
            "set true"
        }
        (if *bool_sig.get() {
            "true"
        } else {
            "false"
        })
    }
}
```

Sycamore 0.8.2 (with Perseus 0.4.0-beta17 or 18), Chrome.

**Workaround:**
Adding wrapping div inside view! removes error:
```rust
#[component]
fn MyComponent<G: Html>(cx: Scope) -> View<G> {
    let bool_sig = create_signal(cx, false);
    view! { cx,
        div {
            button(on:click = |_| {
                bool_sig.set(true);
            }) {
                "set true"
            }
            (if *bool_sig.get() {
                "true"
            } else {
                "false"
            })
        }
    }
}
```

**Comment by wingertge** [comment]
This sounds like exactly like the bug fixed in #573

**Comment by lukechu10** [comment]
Just tested this in 0.9.0-beta.2 and seems to have been fixed by reactivity v3 (#612) so I shall tentatively close this now.
